src/methods/function.js

```diff
@@ -662,7 +663,7 @@ function InternalCall(
   }
 
   // 2. If F's [[FunctionKind]] internal slot is "classConstructor", throw a TypeError exception.
-  if (F.$FunctionKind === "classConstructor")
+  if (F instanceof ECMAScriptFunctionValue && F.$FunctionKind === "classConstructor")
```

Comment:
(Wondering if NativeFunctionValue may also be allowed, not sure how our internal classes relate to the spec.)